### PR TITLE
coverity fix: removing logically dead code

### PR DESCRIPTION
**Pull request by gaurav36**
coverity CID: 1260393

### CODE REVIEW

File: src/osd/osd_types.cc [hunk 1]
```diff
@@ -3606,15 +3606,6 @@ void object_copy_data_t::decode(bufferlist::iterator& bl)
     ::decode(data, bl);
     ::decode(omap_data, bl);
     ::decode(cursor, bl);
-    if (struct_v >= 2)
```
Comment (dachary) [inline src/osd/osd_types.cc]: the if must be removed but not the body

File: src/osd/osd_types.cc [hunk 2]
```diff
@@ -3606,15 +3606,6 @@ void object_copy_data_t::decode(bufferlist::iterator& bl)
...
-      ::decode(omap_header, bl);
-    if (struct_v >= 3) {
-      ::decode(snaps, bl);
-      ::decode(snap_seq, bl);
-    } else {
```
Comment (dachary) [inline src/osd/osd_types.cc]: the else can be removed but not the body of the if

Comment (gaurav36) [inline src/osd/osd_types.cc]: dachary, thank you for reviewing. 

in next patch i will resolve your comments.

**Comment by tchaikov** [comment]
@gaurav36 could you cherry-pick your commit on the master branch, so it only has your commit of https://github.com/gaurav36/ceph/commit/ddb3f913168461b5f873bc58f400f58878bcfd35? and you might want to add a `Signed-off-by:` line at the end of commit log, like

```
Signed-off-by: Gaurav Kumar Garg <ggarg@redhat.com>
```

thanks,

**Comment by tchaikov** [comment]
i am closing this pull request as #3805 was merged into master branch.

@gaurav36 next time, you could simply "git push -f " to the branch of your github repo to update the pull request. so we can preserve the comment, discussion and the history of your patch.

**Comment by gaurav36** [comment]
@tchaikov sure i will do from next time.

**Pull request closed without merging**